packages/apps/autorest/src/core-version-utils.ts

```diff
@@ -0,0 +1,81 @@
+import { EnumerateFiles, ReadUri, ResolveUri } from "@azure-tools/uri";
+import chalk from "chalk";
+import { dirname } from "path";
+import { AutorestArgs } from "./args";
+import { resolveEntrypoint, selectVersion } from "./autorest-as-a-service";
+import * as vm from "vm";
+import { config } from "process";
+
+/**
+ * Return the version requested of the core extension.
+ * @param args ClI args.
+ * @returns npm version/tag.
+ */
+export const getRequestedCoreVersion = (args: AutorestArgs): string => {
+  return args.version || (args.latest && "latest") || (args.preview && "preview") || "latest-installed";
+};
+
+const cwd = process.cwd();
+
+/**
+ * Tries to load the version of autorest core from a config file.
+ * @param args CLI Version
+ * @param selectedVersion 
+ */
+export const configurationSpecifiedVersion = async (args: AutorestArgs, selectedVersion: any) => {
+  try {
+    // we can either have a selectedVerison object or a path. See if we can find the AutoRest API
+    const autorestApi = await resolveEntrypoint(
+      typeof selectedVersion === "string" ? selectedVersion : await selectedVersion.modulePath,
+      "main",
+    );
+
+    // things we need in the sandbox.
+    const sandbox = {
+      require,
+      console,
+      rfs: {
+        EnumerateFileUris: async (folderUri: string): Promise<Array<string>> => {
+          return EnumerateFiles(folderUri, ["readme.md"]);
+        },
+        ReadFile: (uri: string): Promise<string> => ReadUri(uri),
+      },
+      cfgfile: ResolveUri(`file://${cwd}`, args.configFileOrFolder || ""),
```

Comment:
This is pretty much the fix going from 
```
cfgfile: ResolveUri(cwd, args.configFileOrFolder || "."),
```
to
```
cfgfile: ResolveUri(`file://${cwd}`, args.configFileOrFolder || ""),
```

The previous implementation would for `cwd = "/dev/test-folder"` it would return `null:///dev`. So both setting null as the protocol and going up to the parent folder. 

Wondering if this is something that is inconsistent between windows and unix